scientific_library/tvb/simulator/monitors.py

```diff
@@ -248,14 +248,14 @@ def config_for_sim(self, simulator):
                 self.spatial_mask = simulator.surface.region_mapping
             else:
                 conn = simulator.connectivity
-                if self.default_mask[0] == 'cortical':
```

Comment:
From reading the `SpatialAverage.config_for_sim` method, it seem to me that attribute `default_mask` should have 3 options: `region_mapping` (only one accepted in case of Surface simulations), 'cortical' (only for region sim, when the connectivity has this field properly defined), and 'hemispheres' (only for region sim, and again only when this field is properly defined).
Could you try a small refactor in this direction, pls ?

Reply:
I tried something. Please look at my latest changes if they are okay.

Reply:
It looks much better now. Very small things are left to solve (see comments).

Reply:
How about the surface simulations? 
Could you put the "region_mapping" - uneditable value for that use-case?

Reply:
Wouldn't it be better to add a filter there to only allow the RM chosen before?

Reply:
I am not referring to the Projection Monitor `region_mapping ` attribute here, but to the Monitor SpatialAverage.
See my first comment on this thread: if you have a surface level simulation, with `SpacialAverageMonitor`, then `default_mask` can not have values `cortical` or `hemisphere`, but instead should be a 3rd value, available only for SurfaceSimulations: `region_mapping_mask`. At least this is how I understand it from the `SpatialAverage.config_for_sim` method implementation logic. Would you have a different understanding ?